### PR TITLE
Feature/ukr 223 command for current order items

### PR DESCRIPTION
**Pull request by SecondeJK**

### CODE REVIEW

File: database/factories/OrderFactory.php [hunk 1]
```diff
@@ -3,6 +3,8 @@
 /* @var $factory \Illuminate\Database\Eloquent\Factory */
 
 use App\Merchant;
+use App\Order;
+use App\OrderItem;
```
Comment (jump24-luke) [inline database/factories/OrderFactory.php:7]: So below in the class there is a mixture of `App\Order` or `OrderItem`, surely with the `use` statements here we don't need to reference the full namespace below?

Should we also `use Carbon\Carbon` too?

File: tests/Unit/Model/OrderItemTest.php [hunk 2]
```diff
@@ -23,11 +24,59 @@ class OrderItemTest extends TestCase
      */
     protected OrderItem $model;
 
+    /**
+     * @var OrderItemContract|mixed
```
Comment (jump24-luke) [inline tests/Unit/Model/OrderItemTest.php:28]: Remove `|mixed`

File: tests/Unit/Repository/OrderItemRepositoryTest.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+ */
+class OrderItemRepositoryTest extends TestCase
+{
+    use RefreshDatabase;
+    use WithFaker;
+
+    /**
+     * @var OrderContract
+     */
+    protected $repository;
```
Comment (jump24-luke) [inline tests/Unit/Repository/OrderItemRepositoryTest.php:26]: Typehint OrderContract

File: app/Console/Commands/AddVariantIdToOrderItems.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        $this->info('Adding default variant to ' . $ordersToUpdate->count() . ' Order(s)');
+
+        $bar = $this->output->createProgressBar($ordersToUpdate->count());
+
+        $bar->start();
+
+        $ordersToUpdate->each(function (Order $order) use ($bar) {
+            $items = $order->items()->where('variant_id', '=', null)->get();
+            foreach ($items as $item) {
+                $inventory = $item->inventory()->first();
```
Comment (dannewnsjump24) [inline app/Console/Commands/AddVariantIdToOrderItems.php:61]: Can we pre-load these in in the query above to save on N+1

- [ ] preload the relationship in

File: app/Console/Commands/AddVariantIdToOrderItems.php [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                $defaultVariant = $inventory->variants()->first();
```
Comment (dannewnsjump24) [inline app/Console/Commands/AddVariantIdToOrderItems.php:62]: - [ ] Preload relationship

Comment (dannewnsjump24) [reply app/Console/Commands/AddVariantIdToOrderItems.php:62]: We should handle if there is no variant present so do a check if defaultVariant !== null

- [ ] Check around defaultVariant not being null

File: app/Console/Commands/AddVariantIdToOrderItems.php [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        $this->info('Adding default variant to ' . $ordersToUpdate->count() . ' Order(s)');
+
+        $bar = $this->output->createProgressBar($ordersToUpdate->count());
+
+        $bar->start();
+
+        $ordersToUpdate->each(function (Order $order) use ($bar) {
+            $items = $order->items()
+                ->where('variant_id', '=', null)
+                ->with('inventory')
```
Comment (dannewnsjump24) [inline app/Console/Commands/AddVariantIdToOrderItems.php:61]: wont this require a `inventory.variants` to preload that secondary relationship as well?

File: database/factories/OrderFactory.php (hunk 1 above)
Comment (SecondeJK) [reply database/factories/OrderFactory.php:7]: Added in proper aliasing for Carbon and entities

File: app/Console/Commands/AddVariantIdToOrderItems.php (hunk 6 above)
Comment (SecondeJK) [reply app/Console/Commands/AddVariantIdToOrderItems.php:61]: d'oh, I'll add that in

**Pull request merged**